pkg/block/azure/adapter.go

```diff
@@ -191,67 +178,56 @@ func (a *Adapter) Put(ctx context.Context, obj block.ObjectPointer, sizeBytes in
-	uploadOpts.TransferManager = transferManager
-	defer transferManager.Close()
-	resp, err := copyFromReader(ctx, reader, blobURL, uploadOpts)
-	if err != nil {
-		return err
-	}
-	_ = resp == nil // this is done in order to ignore "result 0 is never used" error ( copyFromReader is copied from azure, and we want to keep it with minimum changes)
-	return nil
+	o := a.translatePutOpts(ctx, opts)
+	_, err = a.client.NewContainerClient(qualifiedKey.ContainerName).NewBlockBlobClient(qualifiedKey.BlobURL).UploadStream(ctx, reader, &o)
+	return err
 }
 
 func (a *Adapter) Get(ctx context.Context, obj block.ObjectPointer, _ int64) (io.ReadCloser, error) {
 	var err error
 	defer reportMetrics("Get", time.Now(), nil, &err)
 
-	return a.Download(ctx, obj, 0, azblob.CountToEnd)
+	return a.Download(ctx, obj, 0, blockblob.CountToEnd)
 }
 
 func (a *Adapter) GetPreSignedURL(ctx context.Context, obj block.ObjectPointer, mode block.PreSignMode) (string, error) {
-	permissions := azblob.BlobSASPermissions{Read: true}
+	permissions := aztables.AccountSASPermissions{Read: true}
 	if mode == block.PreSignModeWrite {
-		permissions = azblob.BlobSASPermissions{Write: true}
+		permissions = aztables.AccountSASPermissions{
+			Read:   true,
+			Add:    true,
+			Update: true,
+			Delete: true,
+		}
 	}
 	return a.getPreSignedURL(ctx, obj, permissions)
 }
 
-func (a *Adapter) getPreSignedURL(ctx context.Context, obj block.ObjectPointer, permissions azblob.BlobSASPermissions) (string, error) {
+func (a *Adapter) getPreSignedURL(ctx context.Context, obj block.ObjectPointer, permissions aztables.AccountSASPermissions) (string, error) {
 	qualifiedKey, err := resolveBlobURLInfo(obj)
 	if err != nil {
 		return "", err
 	}
-	vals := azblob.BlobSASSignatureValues{
-		Protocol:      azblob.SASProtocolHTTPS, // Users MUST use HTTPS (not HTTP)
-		ExpiryTime:    a.preSignedURLDurationGenerator(),
-		ContainerName: qualifiedKey.ContainerName,
-		BlobName:      qualifiedKey.BlobURL,
-		Permissions:   permissions.String(),
-	}
 
 	if a.keyCredentials == nil {
 		err = fmt.Errorf("pre-signed mode on Azure is only supported for shared key credentials: %w", ErrNotImplemented)
 		a.log(ctx).WithError(err).Error("no support for pre-signed using this Azure credentials provider")
 		return "", err
 	}
 
-	sasQueryParams, err := vals.NewSASQueryParameters(a.keyCredentials)
+	client, err := aztables.NewServiceClientWithSharedKey(qualifiedKey.ContainerURL, a.keyCredentials, nil)
 	if err != nil {
-		a.log(ctx).WithError(err).Error("could not generate SAS query parameters")
 		return "", err
 	}
-	// Create the URL of the resource you wish to access and append the SAS query parameters.
-	// Since this is a blob SAS, the URL is to the Azure storage blob.
-	qp := sasQueryParams.Encode()
+
+	qp, err := client.GetAccountSASURL(aztables.AccountSASResourceTypes{
+		Container: true,
+		Object:    true,
+	}, permissions, time.Now(), a.preSignedURLDurationGenerator())
```

Comment:
Not sure we should use time.Now() it seems like we pass a time that "IsZero" it isn't specified, which is what I believe we want in this case.